build_product_images.py

```diff
@@ -85,26 +99,55 @@ def build_and_publish_image(args, product):
     """
     Returns a list of commands that need to be run in order to build and
     publish product images.
+
+    For local building, builder instances are supported.
     """
     commands = []
-
+    push = ""
     image_name = f'{args.registry}/stackable/{product["name"]}'
     tags = build_image_tags(image_name, args.image_version, args.product_version)
     build_args = build_image_args(product["versions"][0])
 
-    commands.append(
-        [
-            "docker",
-            "build",
-            *build_args,
-            *tags,
-            "-f",
-            product["name"] + "/Dockerfile",
-            ".",
-        ]
-    )
     if args.push:
-        commands.append(["docker", "push", "--all-tags", image_name])
+        push = "--push"
+
+    # Multiarch builds
+    if args.multiarch:
+        commands.append(
+            [
+                "docker",
+                "buildx",
+                "build",
+                *build_args,
+                *tags,
+                "-f",
+                product["name"] + "/Dockerfile",
+                "--platform",
+                "linux/amd64,linux/arm64",
+                push,
```

Comment:
Is this `push` needed here? The actual pushing of the images is done as an additional (optional) step below. This way I can always build the images locally without pushing them. Needed for testing.

Reply:
It seems to be. When I've tried to separate the comments it happens when you want at least two target platforms, the images without --push won't be loaded to registry, therefore with docker push --all-tags the multi image can not be pushed to the repository. 

To work around that, if -u is not set the variable push will contain an empty string. If -u is given, push contains "--push" and the image will be send to the repository.